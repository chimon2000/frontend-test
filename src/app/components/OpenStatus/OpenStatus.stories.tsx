import * as React from "react";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";
import { action } from "@storybook/addon-actions";
import { withKnobs, boolean, select } from "@storybook/addon-knobs";

import { OpenStatus } from "./OpenStatus";

const stories = storiesOf("Components", module);
stories.add(
  "OpenStatus",
  withInfo({ inline: true })(() => (
    <OpenStatus
      open={boolean("Open", true)}
      size={select("Size", ["normal", "small", "compact"], "normal")}
    />
  ))
);
