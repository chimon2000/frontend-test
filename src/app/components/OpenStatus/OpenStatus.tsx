import * as React from "react";

export interface OpenStatusProps {
  className?: string;
  open: boolean;
  size?: "small" | "normal" | "compact";
}

export const OpenStatus: React.FunctionComponent<OpenStatusProps> = ({
  className,
  size = "normal",
  open
}) => {
  const label = open ? "Open Now" : "Closed";
  return (
    <div className={`open-status open-status--${size}`}>
      <span
        className={`open-status__indicator open-status__indicator--${
          open ? "open" : "closed"
        }`}
      >
        {size === "compact" ? (
          <span className="open-status__compact-label">
            {open ? "Open" : "Closed"}
          </span>
        ) : null}
      </span>
      <span className="open-status__label">{label}</span>
    </div>
  );
};
