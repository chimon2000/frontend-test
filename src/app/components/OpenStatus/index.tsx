export { OpenStatus as default } from "./OpenStatus";
