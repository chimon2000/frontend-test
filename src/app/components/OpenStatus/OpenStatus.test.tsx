import * as React from "react";
import { shallow } from "enzyme";

import { OpenStatus } from "./OpenStatus";

describe("<OpenStatus />", () => {
  it("doesn't blow up", () => {
    const wrapper = shallow(<OpenStatus open />);

    expect(wrapper).toMatchSnapshot();
  });
});
