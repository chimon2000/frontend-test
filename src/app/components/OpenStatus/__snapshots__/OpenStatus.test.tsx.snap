// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OpenStatus /> doesn't blow up 1`] = `
<div
  className="open-status open-status--normal"
>
  <span
    className="open-status__indicator open-status__indicator--open"
  />
  <span
    className="open-status__label"
  >
    Open Now
  </span>
</div>
`;
